src/main/java/com/getindata/connectors/http/internal/table/lookup/LookupQueryInfo.java

```diff
@@ -0,0 +1,60 @@
+package com.getindata.connectors.http.internal.table.lookup;
+
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import lombok.Getter;
+import lombok.ToString;
+
+import com.getindata.connectors.http.internal.utils.uri.NameValuePair;
+import com.getindata.connectors.http.internal.utils.uri.URLEncodedUtils;
+
+/**
+ * Holds the lookup query for an HTTP request.
+ * The {@code  lookupQuery} either contain the query parameters for a GET operation
+ * or the payload of a body-based request.
+ * The {@code bodyBasedUrlQueryParams} contains the optional query parameters of a
+ * body-based request in addition to its payload supplied with {@code  lookupQuery}.
+ */
+@ToString
+public class LookupQueryInfo implements Serializable {
+    @Getter
+    private final String lookupQuery;
+
+    private final Map<String, String> bodyBasedUrlQueryParams;
+
+    public LookupQueryInfo(String lookupQuery) {
+        this(lookupQuery, null);
+    }
+
+    public LookupQueryInfo(String lookupQuery, Map<String, String> bodyBasedUrlQueryParams) {
+        this.lookupQuery =
+                Optional.ofNullable(lookupQuery)
+                        .orElseGet(() -> "");
```

Comment:
although your proposal is equivalent to what I wrote, there would be a performance penalty to use `orElse("") `instead `orElseGet` as explained here: https://www.baeldung.com/java-optional-or-else-vs-or-else-get#orElseGet

Reply:
The link you've provided explains when `orElse` would introduce a performance penalty - that is, when you're calling a method inside the `orElse()` method, especially not straight forward one like `random`. 
This performance hit comes because in case of `orElse` we're always calling that method, no matter if we need it or not. In case of `orElseGet`, we just call it when we actually need it. This is the only difference. Please refer to the implementation of both methods to confirm the above.
And so here, where a constant is used (see Java String Pool), there should be no performance difference whatsoever.

Reply:
Fixed in https://github.com/getindata/flink-http-connector/commit/eba1be71133ccfeeeb020787faa43eac85e85fce#diff-5d0d1d1012fdd57bd2aa4b4fe50dca8d1ce442236c2eea7283222092d917b4a4R35-R40